lib/filterx/func-unset-empties.c

```diff
@@ -46,20 +52,16 @@ static gboolean _process_list(FilterXFunctionUnsetEmpties *self, FilterXObject *
 static gboolean
 _should_unset(FilterXFunctionUnsetEmpties *self, FilterXObject *obj)
 {
-  gsize str_len;
-  const gchar *str;
-  if (filterx_object_extract_string(obj, &str, &str_len))
-    {
-      return str_len == 0 ||
-             strcasecmp(str, "n/a") == 0 ||
-             strcmp(str, "-") == 0;
-    }
-
   if (filterx_object_is_type(obj, &FILTERX_TYPE_NAME(null)))
     {
       return TRUE;
     }
 
+  gsize str_len;
+  const gchar *str;
+  if (filterx_object_extract_string(obj, &str, &str_len) && str_len == 0)
+    return TRUE;
+
   if (filterx_object_is_type(obj, &FILTERX_TYPE_NAME(dict)) ||
```

Comment:
It's a bit strange for me that the user cannot change the behavior of handling `list`s, `dict`s and `null`s.
I don't know if they want to do it, but this seems asymmetric. If as a user I were to set the target strings, and `null`s and empty `dict`s were getting removed, I would be surprised.

Reply:
what would be the proper way to handle that case?